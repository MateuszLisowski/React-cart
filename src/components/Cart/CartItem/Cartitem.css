.singleCartProduct {
    display: flex;
    align-items: center;
    justify-content: center;
    margin: 20px 0;
  }
  .singleCartProduct > div {
    width: 60%;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
  }
  .singleCartProduct > img {
    width: 45%;
    margin: 0;
    height: 100px;
  }