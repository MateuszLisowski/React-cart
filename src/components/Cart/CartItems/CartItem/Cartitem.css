@keyframes enteredAnimation {
  from {
    transform: translateX(-300%);
    z-index: 5; }
  to {
    transform: translateX(0%);
    z-index: 1; } }

.singleCartProduct {
  margin: 20px 0;
  animation: enteredAnimation 0.8s forwards;
  transition: 0.1s transform;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center; }
  .singleCartProduct > div {
    width: 60%;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center; }
  .singleCartProduct > img {
    width: 45%;
    margin: 0;
    height: 100px; }
  .singleCartProduct .removeItem {
    background-color: white;
    color: black;
    border: 1px solid black;
    outline: none;
    box-shadow: 0 8px 16px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
    padding: 15px 32px;
    text-decoration: none;
    font-size: 15px;
    margin: 6px 2px;
    cursor: pointer;
    font-family: monospace; }
    .singleCartProduct .removeItem:hover {
      background-color: #292626;
      color: #f1e3e3;
      border: 1px solid black; }
