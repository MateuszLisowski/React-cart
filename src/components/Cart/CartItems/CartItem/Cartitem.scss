@import '../../../../mixins/mixins.scss';

@keyframes enteredAnimation {
    from {
        @include enterAnimation(translateX(-300%), 5);
    }
    to {
        @include enterAnimation(translateX(0%), 1);
    }
}

.singleCartProduct {
    margin: 20px 0;
    animation: enteredAnimation 0.8s forwards;
    transition: 0.1s transform;
    @include flexConfig(row, center, center);
    >div {
        width: 60%;
        @include flexConfig(column, center, center);
    }
    >img {
        width: 45%;
        margin: 0;
        height: 100px;
    }
    .removeItem {
        @include colorsConfig(white, black, 1px solid black);
        outline: none;
        box-shadow: 0 8px 16px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
        padding: 15px 32px;
        text-decoration: none;
        font-size: 15px;
        margin: 6px 2px;
        cursor: pointer;
        font-family: monospace;
        &:hover {
            @include colorsConfig(#292626, #f1e3e3, 1px solid black);
        }
    }
}