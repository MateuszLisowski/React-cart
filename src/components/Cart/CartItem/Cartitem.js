import React from 'react';
import './Cartitem.css';

const CartItem = ({ product, index, removeItem }) => {
    return (
        <li className='singleCartProduct'>
            <img src='https://imgurbancity.pl/files/sc_staging_images/brand/full_151/694/921/4_03/13406750.jpg' />
            <div>
                <p>
                    {product.name}
                </p>
                <p>
                    {product.price} z≈Ç
                </p>
                <p>
                    chosen size: {product.chosenSize}
                </p>
                <button className='removeItem' onClick={() => removeItem(index)}>
                    Remove item
                </button>
            </div>
        </li>
    )
}

export default CartItem

