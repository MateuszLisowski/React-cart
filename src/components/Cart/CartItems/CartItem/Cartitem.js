import React from 'react';
import './Cartitem.css';
import PropTypes from 'prop-types';

const CartItem = ({ product, index, removeItem }) => {
    return (
        <li className='singleCartProduct'>
            <img src='https://imgurbancity.pl/files/sc_staging_images/brand/full_151/694/921/4_03/13406750.jpg' alt='product' />
            <div>
                <p>
                    {product.name}
                </p>
                <p>
                    {product.price} z≈Ç
                </p>
                <p>
                    chosen size: {product.chosenSize}
                </p>
                <button className='removeItem' onClick={() => removeItem(index)}>
                    Remove item
                </button>
            </div>
        </li>
    )
}

CartItem.propTypes = {
    product: PropTypes.object.isRequired, 
    index: PropTypes.number.isRequired, 
    removeItem: PropTypes.func.isRequired
};

export default CartItem

