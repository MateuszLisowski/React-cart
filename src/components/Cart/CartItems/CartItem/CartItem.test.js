import React from 'react';
import CartItem from './Cartitem';
import renderer from 'react-test-renderer';

describe('<CartItem />', () => {
    it('renders correctly', () => {
        const tree = renderer
          .create(<CartItem index={1} product={{name: 'nike', price: 22, chosenSize: 2}} removeItem={() => {}}/>)
          .toJSON();
        expect(tree).toMatchSnapshot();
      });
})
