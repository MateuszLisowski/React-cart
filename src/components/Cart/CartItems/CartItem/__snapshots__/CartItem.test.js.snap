// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartItem /> renders correctly 1`] = `
<li
  className="singleCartProduct"
>
  <img
    alt="product"
    src="https://imgurbancity.pl/files/sc_staging_images/brand/full_151/694/921/4_03/13406750.jpg"
  />
  <div>
    <p>
      nike
    </p>
    <p>
      22
       z≈Ç
    </p>
    <p>
      chosen size: 
      2
    </p>
    <button
      className="removeItem"
      onClick={[Function]}
    >
      Remove item
    </button>
  </div>
</li>
`;
